.eslintrc.js

```diff
@@ -125,13 +127,39 @@ module.exports = {
      * TypeScript files
      */
     {
-      files: ['*.{ts,tsx}'],
+      files: tsconfig.fileNames.filter((f) => /\.tsx?$/u.test(f)),
```

Comment:
Our tsconfig technically includes files other than `.tsx` (it does NOT type check them though), but we previously didn't apply the typescript eslint rules to those files (they won't come close to passing) so I'm filtering them out here.